keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepository.kt

```diff
@@ -162,6 +163,17 @@ interface ArtifactRepository : PeriodicallyCheckedRepository<DeliveryArtifact> {
     targetEnvironment: String
   )
 
+  /**
+   * Marks a version of an artifact as skipped for an environment, with information on what version skipped it.
+   */
+  fun markAsSkipped(
+    deliveryConfig: DeliveryConfig,
+    artifact: DeliveryArtifact,
+    version: String,
+    targetEnvironment: String,
+    skippedByVersion: String
```

Comment:
```suggestion
    supersededByVersion: String
```

Reply:
Oh, this is better wording. Let me see how to work this in.